presto-jdbc/src/test/java/io/prestosql/jdbc/TestJdbcWarnings.java

```diff
@@ -122,69 +134,47 @@ public void testStatementWarnings()
+                        "SELECT * FROM blackhole.blackhole.slow_table");
                 queryCreationFuture.set(null);
             }
             catch (Throwable e) {
                 queryCreationFuture.setException(e);
             }
         });
-        while (statement.getWarnings() == null) {
-            Thread.sleep(100);
-        }
-        SQLWarning warning = statement.getWarnings();
-        Set<WarningEntry> currentWarnings = new HashSet<>();
-        assertTrue(currentWarnings.add(new WarningEntry(warning)));
-        for (int warnings = 1; !queryCreationFuture.isDone() && warnings < 100; warnings++) {
-            for (SQLWarning nextWarning = warning.getNextWarning(); nextWarning == null; nextWarning = warning.getNextWarning()) {
-                // Wait for new warnings
-            }
-            warning = warning.getNextWarning();
-            assertTrue(currentWarnings.add(new WarningEntry(warning)));
-            Thread.sleep(100);
-        }
-        assertEquals(currentWarnings.size(), 100);
+
+        Set<WarningEntry> currentWarnings = getStatementWarnings(statement, queryCreationFuture);
+        assertThat(currentWarnings).size().isGreaterThanOrEqualTo(PRELOADED_WARNINGS + 1);
+
         queryExecutor.shutdownNow();
     }
 
     @Test
     public void testLongRunningQuery()
-            throws SQLException, InterruptedException
+            throws Exception
     {
         ExecutorService queryExecutor = newSingleThreadExecutor(daemonThreadsNamed("test-%s"));
         QueryCreationFuture queryCreationFuture = new QueryCreationFuture();
         queryExecutor.submit(() -> {
             try {
-                statement.execute("SELECT 1 AS col1 FROM tpch.sf1.lineitem CROSS JOIN tpch.sf1.lineitem");
+                statement.execute("SELECT * FROM blackhole.blackhole.slow_table");
                 queryCreationFuture.set(null);
             }
             catch (Throwable e) {
                 queryCreationFuture.setException(e);
             }
         });
-        while (statement.getResultSet() == null) {
-            Thread.sleep(100);
-        }
-        ResultSet resultSet = statement.getResultSet();
-        Set<WarningEntry> currentWarnings = new HashSet<>();
-        for (int rows = 0; !queryCreationFuture.isDone() && rows < 10; ) {
-            if (resultSet.next()) {
-                for (SQLWarning warning = resultSet.getWarnings(); warning.getNextWarning() != null; warning = warning.getNextWarning()) {
-                    assertTrue(currentWarnings.add(new WarningEntry(warning.getNextWarning())));
-                }
-            }
-            else {
-                break;
-            }
-            Thread.sleep(100);
-        }
+
+        Set<WarningEntry> currentWarnings = getStatementWarnings(statement, queryCreationFuture);
+        assertThat(currentWarnings).size().isGreaterThanOrEqualTo(PRELOADED_WARNINGS + 1);
+
         queryExecutor.shutdownNow();
```

Comment:
This should be `finally`

Reply:
Agreed, though I think that makes the code harder to read. I'll move it to the setup/tearndown.